File: C/bubble-sort.c
#include <stdio.h>
#include<stdlib.h>

void swapper(int *p1,int *p2){      //function to perform swapping
    int temp= *p1;
    *p1=*p2;
    *p2=temp;
}

void bubbleSort(int a[],int size){     //function to perform bubble sort
    int i,j;

    for(int i=0;i<size-1;i++){
        for(j=0;j<size-i-1;j++){
            if(a[j]>a[j+1])
                swapper(&a[j],&a[j+1]);
        }
    }
}

void display(int a[],int size){
    int i;                                             //function to display the sorted array
    printf("\nThe sorted array is ");
    for(i=0;i<size;i++){
        printf("%d ",a[i]);
    }
}

int main(){
    int size,i;

    printf("Enter the number of elements ");   //size of array input from user
    scanf("%d",&size);

    int *a=(int*)calloc(size,sizeof(int));  //memory allocated for array dynamically

    printf("\nEnter the elements ") ;       //input of elements by the user
    for(i=0;i<size;i++){
        scanf("%d",&a[i]);
    }

    bubbleSort(a,size);

    display(a,size);


    return 0;
}

File: Algorithms/Sorting/bubbleSort.java
import java.util.Scanner;

public class bubbleSort {

    public static void main(String[] args) {
        int num, i, j, temp;
        Scanner input = new Scanner(System.in);
        System.out.println("Enter the number of Soldiers:");
        num = input.nextInt();
        int array[] = new int[num];
        System.out.println("Enter " + num + " soldiers heights in centrimeters(cm): ");
 for (i = 0; i < num; i++) {
            array[i] = input.nextInt();
        }
        for (i = 0; i < (num - 1); i++) {
            for (j = 0; j < num - 1 - i; j++) {
                if (array[j + 1] < array[j]) {
                    temp = array[j];
                    array[j] = array[j + 1];
                    array[j + 1] = temp;
                }
            }
        }
        System.out.println("Sorted list of soldiers heights:");
        for (i = 0; i < num; i++) {
            System.out.println(array[i]);
        }
    }
}
